File: recursion/count_digits.c
#include <stdio.h>

int count_digits(int num) {
	if (num == 0)
		return 0;
	return 1 + count_digits(num / 10);
}

void run(int num) {
	printf("digits: %d\t%d\n", num, count_digits(num));
}

int main() {
	run(0);
	run(10);
	run(432);
	run(40948309);
	run(4390);
	return (0);
}

File: recursion/count_digits.js
/**
 * @param {number} num
 * @returns {number}
 */
function countDigits(num) {
	if (num == 0)
		return 0;
	return 1 + countDigits(Math.floor(num / 10));
}

function run(num) {
	console.log(`digits: ${num}\t${countDigits(num)}`);
}

run(0);
run(10);
run(483);
run(49302490);
run(8934);
